files/cloud-init/nfs/cloud-config

```diff
@@ -56,8 +56,13 @@ runcmd:
   #
   # Update /etc/exports - NOTE: The CIDR provided works for the whole VPC
   #
-  - echo "/export         ${aks_cidr_block}(rw,no_root_squash,async,insecure,fsid=0,crossmnt,no_subtree_check)" >> /etc/exports
-  - echo "/export         ${misc_cidr_block}(rw,no_root_squash,async,insecure,fsid=0,crossmnt,no_subtree_check)" >> /etc/exports
+  - if [ ${aks_cidr_block} == ${misc_cidr_block} ]; 
+  - then 
+  -   echo "/export         ${aks_cidr_block}(rw,no_root_squash,async,insecure,fsid=0,crossmnt,no_subtree_check)" >> /etc/exports;
+  - else
+  -   echo "/export         ${aks_cidr_block}(rw,no_root_squash,async,insecure,fsid=0,crossmnt,no_subtree_check)" >> /etc/exports && \
+  -   echo "/export         ${misc_cidr_block}(rw,no_root_squash,async,insecure,fsid=0,crossmnt,no_subtree_check)" >> /etc/exports;
+  - fi
```

Comment:
Would the for loop example not work here? Was wondering if that could be implemented as we have this on the AWS side. I have not checked on the GCP side. Was thinking that the combined items since they were the same would only yield 1 entry. Can you test this?

Reply:
Not sure how to test the whole scenario but I tried testing the for loop logic separately. For loop doesn't seem to combine the items, treats it as a list and iterates over all the items. We would need some extra logic to make it a unique list for the 'for' loop to iterate over.
Tbh if-else seems much simpler to accomplish this task.

Reply:
Just would need to test when both networks are the same. If this is no feasible, then the code base works fine with the updated if statement.